### PR TITLE
How to Auto Log Off Users Due to Inactivity Using Angular 12

### LINKED ISSUE #3406: How to Auto Log Off Users Due to Inactivity Using Angular 12
### Proposed title of article
How to Auto Log Off Users Due to Inactivity Using Angular 12

### Introduction paragraph (2-3 paragraphs):
An application I’ve been building has some very robust security requirements, one of which is a client side session expiration. This is a feature you’ve likely noticed when logging into mobile banking. You log into your account, switch to another tab, and return to that tab several minutes later to find that you’ve been logged out for inactivity. This helps prevent sensitive information from remaining on an idle user’s screen and reduces the likelihood of a malicious user from accessing the application.

The use case seemed common enough. I started scouring the internet for a solution using Angular 2+ and Ngrx. Alas, I struggled to find a clean solution and went on trying to roll my own. It turned out that implementing this is  simple and I wanted to share it with the interwebs.

### Key takeaways: 
1.Angular 12+ session expiry solution
2. Browsers events handling
3. Monitoring user inactivity on the site.

### PR DESCRIPTION
**Pull request by owinowendy**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3406

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
plagiarism check done ✅

**Comment by jumamiller** [comment]
@owinowendy, kinldy add the hero image. Thanks.

**Review by jumamiller — APPROVED** [review]
Hello @owinowendy , kindly add an hero image.

**Comment by owinowendy** [comment]
@jumamiller, uploaded!

**Review by jumamiller — APPROVED** [review]

**Comment by Kayere** [comment]
Hello @owinowendy, this is a nice article, seems to be very helpful. However, your article's word count (Excluding code snippets) is less than 750. Could you please add more content so as to meet the minimum word count.

Ping me when you are done so that we can publish your article.

**Comment by owinowendy** [comment]
> Hello @owinowendy, this is a nice article, seems to be very helpful. However, your article's word count (Excluding code snippets) is less than 750. Could you please add more content so as to meet the minimum word count.
> 
> Ping me when you are done so that we can publish your article.

@Kayere, updates done! Thanks

File: content/articles/auto-logout-angular/index.md [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+	text-decoration-line: underline;
+	color: #ffffff;
+}
+```
+
+We're editing the default Angular material form with our defined CSS styles in the styles file above. Of course, you're free to customize this to meet your needs.
+
+Output:
+
+![register](/engineering-education/auto-logout-angular/register.png)
```
Comment (Kayere) [inline content/articles/auto-logout-angular/index.md:347]: Please upload this image.

**Comment by owinowendy** [comment]
@Kayere, uploaded

**Pull request merged**